File: 프로그래머스/3/43162. 네트워크/네트워크.py
def dfs(n, computers, i):
    visited[i] = True
    
    for j in range(n):
        if not visited[j] and computers[i][j] == 1:
            dfs(n, computers, j)

def solution(n, computers):
    global visited
    
    answer = 0
    visited = [False] * n
    
    for i in range(n):
        if not visited[i]:
            answer += 1
            dfs(n, computers, i)
            
    return answer

File: 프로그래머스/3/43162. 네트워크/네트워크.java
class Solution {
    public int solution(int n, int[][] computers) {
        boolean[] visited = new boolean[n];
        int answer = 0;

        for (int i = 0; i < n; i++) {
            if (!visited[i]) {
                dfs(visited, i, n, computers);
                answer++;
            }
        }

        return answer;
    }
    
    public void dfs(boolean[] visited, int start, int n, int[][] computers) {
        visited[start] = true;

        for (int i = 0; i < n; i++) {
            if (!visited[i] && computers[start][i] == 1) {
                dfs(visited, i, n, computers);
            }
        }
    }
}